### PR TITLE
III-5124 Multiple subevents same day

### PR DESCRIPTION
**Pull request by JonasVHG**
### Fixed

- CalendarSummary format `md` with multiple subEvents on the same day will no longer duplicate the dates

---

Ticket: https://jira.uitdatabank.be/browse/III-5124

### CODE REVIEW

**Comment by lucwollants** [comment]
I wonder if this "issue" could also happen in other calendar formats.

**Comment by JonasVHG** [comment]
> 



> I wonder if this "issue" could also happen in other calendar formats.
I've tried to create such an event but a single CalenderType and a Permanent only have 1 start & enddate. And a permanent has none. So I think it's only possible on Multiple.

**Comment by bertramakers** [comment]
Only calendarType "multiple" can have multiple subEvents indeed, but I suspect that @LucWollants meant other formats like "sm" and "lg" in addition to "md"

**Comment by JonasVHG** [comment]
> Only calendarType "multiple" can have multiple subEvents indeed, but I suspect that @LucWollants meant other formats like "sm" and "lg" in addition to "md"

No they do not,  eg. https://io.uitdatabank.be/event/cbb5e4bf-4948-4348-8b2f-826355f7524c/calsum?format=xs 
https://io.uitdatabank.be/event/cbb5e4bf-4948-4348-8b2f-826355f7524c/calsum?format=sm
https://io.uitdatabank.be/event/cbb5e4bf-4948-4348-8b2f-826355f7524c/calsum?format=lg (because hours are different)
vs. https://io.uitdatabank.be/event/cbb5e4bf-4948-4348-8b2f-826355f7524c/calsum?format=md

**Comment by bertramakers** [comment]
> No they do not, eg.

Thanks for clarifying with some examples!

File: src/Multiple/MediumMultipleHTMLFormatter.php [hunk 1]
```diff
@@ -38,6 +38,7 @@ public function format(Offer $offer): string
                 $subEventSummaries[] = $formatter->format($subEvent);
             }
         }
+        $subEventSummaries = array_unique($subEventSummaries);
```
Comment (bertramakers) [inline src/Multiple/MediumMultipleHTMLFormatter.php:41]: Clever 👍

**Review by bertramakers — APPROVED** [review]

**Pull request merged**